urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/PairsInAv231.java

```diff
@@ -0,0 +1,243 @@
+package permlib.examples;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Objects;
+import permlib.PermUtilities;
+import permlib.Permutation;
+import permlib.Permutations;
+import permlib.Symmetry;
+import permlib.classes.PermutationClass;
+import permlib.processor.PermCounter;
+import permlib.property.HereditaryProperty;
+
+/**
+ * Looking for equivalences in Av(231, p, q)
+ *
+ * @author Michael Albert
+ */
+public class PairsInAv231 {
+
+    public static void main(String[] args) {
+        doWilf(3,4, true, 10);
+        // doSpecial();
+    }
+
+    public static void doSpecial() {
+
+        PermutationClass c = new PermutationClass("231", "321");
+        long[] cCounts = new long[8];
+        for (int n = 4; n < 12; n++) {
+            PermCounter counter = new PermCounter();
+            c.processPerms(n, counter);
+            cCounts[n - 4] = counter.getCount();
+        }
+        System.out.println(Arrays.toString(cCounts));
+        c = new PermutationClass("231", "123");
+        for (int n = 4; n < 12; n++) {
+            PermCounter counter = new PermCounter();
+            c.processPerms(n, counter);
+            cCounts[n - 4] = counter.getCount();
+        }
+        System.out.println(Arrays.toString(cCounts));
+        c = new PermutationClass("231", "321");
+        for (Permutation q : new Permutations(c, 4)) {
+            PermutationClass d = new PermutationClass("231", "321", q.toString());
+            for (int n = 4; n < 12; n++) {
+                PermCounter counter = new PermCounter();
+                d.processPerms(n, counter);
+                cCounts[n - 4] = counter.getCount();
+            }
+            System.out.println(q + " " + Arrays.toString(cCounts));
+        }
+    }
+
+    public static void doWilf(int n, boolean verbose, int checkTotal) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 59-65, lines 66-73 into separate methods.